src/JustSaying/AwsTools/MessageHandling/ISqsQueue.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.SQS.Model;
+
+namespace JustSaying.AwsTools.MessageHandling
+{
+    public interface ISqsQueue
+    {
+        string QueueName { get; }
+
+        string RegionSystemName { get; }
+
+        Uri Uri { get; }
+
+        Task<IList<Message>> GetMessages(int count, List<string> requestMessageAttributeNames, CancellationToken cancellationToken);
+
+        Task<ReceiveMessageResponse> GetMessages(ReceiveMessageRequest request, CancellationToken cancellationToken);
```

Comment:
Agree here, we'll give this a think and try and re-arrange things.

Reply:
There are 2 methods here because SqsNotificationListener uses one and the ConsumerBus uses another.
I'm currently in the process of porting the SqsNotificationListener tests into a ConsumerBus equivalent, and will then remove the SqsNotificationListener and the extra method.

Reply:
This is now gone, thanks!